SPECS/boost/boost.spec

```diff
@@ -1,15 +1,15 @@
-Summary:	Boost
-Name:		boost
-Version:	1.66.0
-Release:        3%{?dist}
-License:	Boost Software License V1
-URL:		http://www.boost.org/
-Group:		System Environment/Security
+%define underscore_version %(echo %{version} | cut -d. -f1-3 --output-delimiter="_")
+Summary:        Boost
+Name:           boost
+Version:        1.82.0
```

Comment:
Since this is a fairly substantial upgrade, I did a grep for boost within 1.0/SPECS and found the following packages would potentially be impacted:
- auoms
- azure-storage-cpp
- blobfuse
- bond
- ceph
- cpprest
- hyperscan
- msgpack
- swig
- systemtab
- toml11
- websocketpp

I ran a buddy build to make sure these packages built properly following the upgrade:
AMD64: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=347146&view=results
ARM64: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=347145&view=results